src/send/usePrepareSendTransactions.ts

```diff
@@ -0,0 +1,71 @@
+import { useState } from 'react'
+import {
+  PreparedTransactionsResult,
+  prepareERC20TransferTransaction,
+} from 'src/viem/prepareTransactions'
+import { TokenBalance, tokenBalanceHasAddress } from 'src/tokens/slice'
+import BigNumber from 'bignumber.js'
+import { useAsyncCallback } from 'react-async-hook'
+import { tokenSupportsComments } from 'src/tokens/utils'
+import Logger from 'src/utils/Logger'
+import { tokenAmountInSmallestUnit } from 'src/tokens/saga'
+
+const TAG = 'src/send/usePrepareSendTransactions'
+
+// just exported for testing
+export async function _prepareSendTransactionsCallback({
+  amount,
+  token,
+  recipientAddress,
+  walletAddress,
+  isDekRegistered,
+  feeCurrencies,
+}: {
+  amount: BigNumber
+  token: TokenBalance
+  recipientAddress: string
+  walletAddress: string
+  isDekRegistered: boolean
+  feeCurrencies: TokenBalance[]
+}) {
+  const includeRegisterDekTx = !isDekRegistered && tokenSupportsComments(token)
+
+  if (amount.isLessThanOrEqualTo(0)) {
+    return
+  }
+  if (tokenBalanceHasAddress(token)) {
+    if (!includeRegisterDekTx) {
+      return prepareERC20TransferTransaction({
+        fromWalletAddress: walletAddress,
+        toWalletAddress: recipientAddress,
+        sendToken: token,
+        amount: BigInt(tokenAmountInSmallestUnit(amount, token.decimals)),
+        feeCurrencies,
+      })
+    }
+    // TODO(ACT-955): case where we need to register DEK too
+  }
+  // TODO(ACT-956): non-ERC20 native asset case
+}
+
+export function usePrepareSendTransactions(
+  prepareSendTransactionsCallback = _prepareSendTransactionsCallback // just for testing
+) {
```

Comment:
nit: I'm not entirely sold on exposing this. I get that it may make testing easier, but also in a way it adds unnecessary details to the interface of the hook which may confuse consumers. 🤔 

Or maybe we should prefix these parameters with `_` and add jsdoc comments so it's clear?

Reply:
I always feel like there is a faustian bargain between writing a proper unit test and avoding "just for testing" DI. I feel gross in either case. Most of the time, I try to move the helper function into its own module so I can mock it and have it both ways (no ugly DI, and single-purpose tests). But in this case, the prepare send transactions callback really did not make sense to put in its own module.

Reply:
I took your suggestion with the jsdoc and parameter rename

Reply:
now that usePrepareSendTransactions has slimmed down, I am less averse to testing it without mocking _prepareSendTransactionsCallback. updated.